gatsby-node.js

```diff
@@ -381,6 +381,7 @@ exports.createPages = async ({ actions, graphql, reporter }) => {
             path: node.fields.slug,
             component: template,
             context: {
+              dirname: path.parse(node.fields.slug).dir,
```

Comment:
Instead of passing in the `dirname` as part of page context you can derive it from `slug` in courses.js. That way it only derived when needed.